### DISCUSSION TITLE
Visuals suggestions

### DISCUSSION THREAD

**Discussion by InterClaw** [fayaz12g/totk-aar#46] [Ideas]
I've got a few suggestions to this tab:

1. Perhaps the custom resolution boxes should trade places. 1920x1080 makes more sense than 1080x1920. 😃 I tried filling them in the logical way and the game looked weird. When filling them in the backwards way they are now it worked correctly.
2. Could you include the mod "LOD Improvement" as well?

![image](https://github.com/fayaz12g/totk-aar/assets/48981208/a41985ed-b592-48df-a7d9-84405ba92380)

**Comment by fayaz12g** [comment]
1) I originally had it as width x height but for some reason switched the order, I forget why. Anyways, I’ve reverted it so now it’s more logical with width x height! The reason it didn’t work when the numbers were entered reversed is the dFPS mod has specific entries for width and height separately which are filled in by what’s entered. 

8.0.1 incorporates that addition, thanks for the suggestion!

2) If I’m not mistaken, the LOD Improvement is just the Disabled LOD Reduction? I pulled it from the GBAtemp repository visual fixes. Unless you mean something else.

**Comment by InterClaw** [comment]
1.
Makes sense now. Thanks! The tab order is wrong now though, hehe. 🙃 

2.
No, they seem to be separate mods. And "Disable LOD Reduction" was renamed to "Disable Quality Reduction", I believe. Perhaps in an effort to reduce the confusion between the two?

So here's the background to this. I've been using your mod pretty much exclusively for the combination of the aspect ratio mod and the controller mod. I haven't been using the visual/graphical mods that you've included so far. For those mods I've been using the collection from here:
https://github.com/HolographicWings/TOTK-Mods-collection

Until now. With the new dFPS mod, which seem to be including a lot more than before, like the higher rendered resolutions etc., there aren't that many other mods you actually need anymore!

Therefore, it's quickly becoming very convenient to aim to ONLY be using your mod - if it can combine everything that's needed for the overall visual experience. 👍 

Their README file has some information on the different mods included in their collection (perhaps check out what these two mods do there!), but for a more up to date instruction covering the latest dFPS version, there's the SETUP file: https://github.com/HolographicWings/TOTK-Mods-collection/blob/main/SETUP.md

Apparently, what you're supposed to be using now are the three dFPS mods + "Disable Quality Reduction" & "LOD Improvement". And aspect ratio and controller mods of course. 😃 

I'm not sure who's to be given credit for each of those two mods, but it seems to be ChucksFeedAndSeed for the former, but I have no idea about the latter.

Truth be told, I haven't gone through [the thread of the 1.5.5 betas here](https://www.reddit.com/user/ChucksFeedAndSeed/comments/14xksce/beta1_totk_dynamicfps_v155beta1_custom_internal/). It's kind of hard to get a clear idea from there what you're supposed to find and download where and run with what and how. I don't think there's a GitHub page for Chucks' stuff. So I'm basing all this on the SETUP guide above.

... so to summarize, I think it would be great if you could include "LOD Improvement" as an option in the list! (And perhaps rename the other one to "Disable Quality Reduction".)



Let me ask you this, the visual/graphical mods you include, are the latest versions of those pulled by the tool when you generate? Or do you have your own copies of those mods, so to speak, so _you_ first have to update to them before the tool can provide them?

**Comment by fayaz12g** [comment]
Hey! Thanks for all the explanations, I've now updated the LOD Improvement mod! I was confused as previously I had the Disable Quality Reduction named as disable LOD Reduction (as you mentioned) but I've changed that in 8.1.0. I've also added the missing options from that repository, including increase camera speed, remove lens flare, and Sky Island Fix (which with dFPS isn't necessary anymore, but I’ve added it in case someone still wants to use Yuzu scalers). I've also brought back the previous static fps and shadow res options if for whatever reason someone doesn't want dfps. Eventually it will be separated into legacy visuals but for now it is just a visuals 2 tab with these new additions. 

As for the dFPS version, I’ve been following that thread actively and bringing in the latest betas when he releases them (currently beta 5)

To answer your last question, because of the way the mod is generated, it is not automatically pulled from the repo but rather I manually have to update the tool every time there's updates to those. I initially did this because I just used the mods.zip from the GBAtemp collection, and haven't ever touched the HolographicWings collection. The way the exefs is generated is all one file for everything, with comments explaining each part. If I were to automatically pull the latest versions of these mods, it would clutter a mod folder with dozens of mods for each checkbox, so I manually coded them into visuals.py to make it easier on the user.  

Oh also, what do you mean by the tab order is wrong now?

Lastly, I wanted to say I plan on adding a 1 click button for visuals that selects all the best options based on their screen size and aspect ratio and GPU (for setting resolution) to make it even easier for the user

**Comment by InterClaw** [comment]
It all sounds really good! Thank you for keeping us up to date. 👍 

Tab order? Ah, it's just a minor thing. 😄 When you press tab to jump between the text boxes, you end up in the right custom resolution box before the left one now. It's probably just because you had them the other way around before and you originally added them in that order. I think it's just a property on the text box. (Not a developer, hehe.)

**Reply by fayaz12g** [reply]
Of course! And I see, thank you for finding that, it's a minor thing but it's an easy fix that would otherwise be overlooked :). If you want to know, it isn't a property on the entry field (textbox) but rather the order they are coded in-- I switch the order they are "packed", and thus displayed on the page, but they were still coded in the reverse order. For example, before it was

height
width

show height
show width

and I had changed it to

height
width

show width
show height

I've now properly updated the code to be

width
height

show width
show height

and it is fixed in 8.1.3
